File: PROYECTO 1/worker.cpp
#include "worker.h"

using namespace std;

// Constructor de la clase Worker
Worker::Worker(int _id, const string& _name, const string& _lastName,
    const string& _birthdate, const string& _gender,
    const string& _address, int _phoneNumber, const string& _email,
    const string& _position, int _yearsOfExperience, int _maxDailyHours)
    : Person(_id, _name, _lastName, _birthdate, _gender, _address, _phoneNumber, _email),
    position(_position), yearsOfExperience(_yearsOfExperience), maxDailyHours(_maxDailyHours) {
}

// Métodos get de Worker
string Worker::getPosition()  { return position; }
int Worker::getYearsOfExperience()  { return yearsOfExperience; }
int Worker::getMaxDailyHours()  { return maxDailyHours; }

// Métodos set de Worker
void Worker::setPosition(const string& _position) { position = _position; }
void Worker::setYearsOfExperience(int _yearsOfExperience) { yearsOfExperience = _yearsOfExperience; }
void Worker::setMaxDailyHours(int _maxDailyHours) { maxDailyHours = _maxDailyHours; }

File: pythonPr/worker.py
from person import Person

class Worker(Person):
    def __init__(self,  _ident = None,  _name = None,  _lastName = None,  _birthdate = None,  _gender = None,  _address = None,  _phoneNumber = None,  _email = None,  _position = None,  _yearsOfExperience = None,  _maxDailyHours = None):
        super().__init__(_ident, _name, _lastName, _birthdate, _gender, _address, _phoneNumber, _email)
        self.position = _position
        self.yearsOfExperience = _yearsOfExperience
        self.maxDailyHours = _maxDailyHours

    # Métodos get de Worker
    def getPosition(self):
        return self.position

    def getYearsOfExperience(self):
        return self.yearsOfExperience

    def getMaxDailyHours(self):
        return self.maxDailyHours

    # Métodos set de Worker
    def setPosition(self, _position):
        self.position = _position

    def setYearsOfExperience(self, _yearsOfExperience):
        self.yearsOfExperience = _yearsOfExperience

    def setMaxDailyHours(self, _maxDailyHours):
        self.maxDailyHours = _maxDailyHours
